### DISCUSSION TITLE
Lab 07.01 - Task 3

### DISCUSSION THREAD

**Discussion by ThomasPineda** [rramosp/ai4eng.v1#224] [07 - DISEÑO DE ALGORITMOS]
Estoy un poco perdido con está tarea, no entiendo como tenemos que llamar a los parametros X, t y b en la función loss, así lo hice yo y al correr la función me sale que la optimización siempre los parametros seran 0.
No sé si estoy poniendo mala la función loss dentro de mi código
![image](https://user-images.githubusercontent.com/86691796/219884766-699d5eff-4fb6-4795-b6fb-d20182f2e861.png)

`from os import X_OK
def fit(X,y):
    from scipy.optimize import minimize
    

    def loss(params):
        X = params[0]
        t = params[1]
        b = params[2]
        sigm = lambda z: 1/(1+np.exp(-z))        
        return np.sum(((sigm(np.dot(X,t) + b))-y)**2)* (1/200)
    
    tinit = np.zeros(shape=X.shape[1]+1) # keep this initialization
    
    r = minimize(fun= loss,x0=tinit, method='Nelder-Mead', tol=1e-5) # complete the call to minimize
    t = r.x[0:2] # extract t from r.x
    b = r.x[-1:] # extract b from r.x
    return t, b`

Anexo el colab: 

https://colab.research.google.com/drive/1Ss1Z-Dgvq6JlvcgZCjYp0xt_6ZVzW7yR?usp=sharing

**Comment by Felipe-RA** [comment]
Saludos  @ThomasPineda 

Debes extraer `t` y `b` de la variable `params`
 
```
 def loss(params):
------> X = params[0]
    t = params[1]
    b = params[2]
    sigm = lambda z: 1/(1+np.exp(-z))        
    return np.sum(((sigm(np.dot(X,t) + b))-y)**2)* (1/200)

```

En la línea que te señalo con la flecha, estas sobreescribiendo el valor de X con params[0] (Preguntate qué hay en `params` al comienzo de `def loss()`)

Adicionalmente, `t` debe obtener los primeros dos elementos, no sólo el primero.


```
 def loss(params):

    t = params[:2]
    b = params[-1]
    sigm = lambda z: 1/(1+np.exp(-z))        
    return np.sum(((sigm(np.dot(X,t) + b))-y)**2)* (1/200)

```

**Comment by imEag** [comment]
Hola! así como dice Felipe, la variable t y b deben salir del parámetro params. Recuerda que en el task 2 te dicen la forma que debe tener cada variable para la función loss y en el task 3 debes implementarlo pero ajustandolo para recibir t y b en una lista y luego extraerlos de ella.
